src/private.rs

```diff
@@ -87,6 +95,14 @@ impl Mode for Emit {
         f(x)
     }
     #[inline(always)]
+    fn map_with_ctx<T, U, C, F: FnOnce(T, C) -> U>(
+        x: Self::Output<T>,
+        ctx: C,
+        f: F,
+    ) -> Self::Output<U> {
+        f(x, ctx)
+    }
```

Comment:
I think you should be able to do this with just `M::map(x, move |x| f(x, c))`, so no need to add another method to `Mode`.